src/katello/tracer.py

```diff
@@ -22,8 +38,13 @@ def upload_tracer_profile(queryfunc, plugin=None):
 
 
 def query_affected_apps(plugin=None):
-    query = Query()
-    return query.affected_applications().get()
+    if yum or dnf:
```

Comment:
Here's pseudocode for another way to write this:

```
if Query then
 Query().affected_applications().get()
else
 collect_apps()
```

That way this method doesn't need to be so concerned with package managers. What do you think?

Reply:
I would prefer my implementation because:
- it tries to get exactly what it is needed (yum, apt or zypp)
- its possible to raise an exception if something is really wrong and its not possible to get the state.